pkg/testing/ca/ca.go

```diff
@@ -56,16 +58,17 @@ import (
 )
 
 type VirtualSigstore struct {
-	fulcioCA              *root.FulcioCertificateAuthority
-	fulcioIntermediateKey *ecdsa.PrivateKey
-	tsaCA                 *root.SigstoreTimestampingAuthority
-	tsaLeafKey            *ecdsa.PrivateKey
-	rekorKey              *ecdsa.PrivateKey
-	ctlogKey              *ecdsa.PrivateKey
-	publicKeyVerifier     map[string]root.TimeConstrainedVerifier
+	fulcioCA                *root.FulcioCertificateAuthority
+	fulcioIntermediateKey   *ecdsa.PrivateKey
+	tsaCA                   *root.SigstoreTimestampingAuthority
+	tsaLeafKey              *ecdsa.PrivateKey
+	rekorKey                *ecdsa.PrivateKey
+	ctlogKey                *ecdsa.PrivateKey
+	publicKeyVerifier       map[string]root.TimeConstrainedVerifier
+	signingAlgorithmDetails signature.AlgorithmDetails
 }
 
-func NewVirtualSigstore() (*VirtualSigstore, error) {
+func NewVirtualSigstoreCustom(signingKeyDetails v1.PublicKeyDetails) (*VirtualSigstore, error) {
```

Comment:
nit on the name to be more descriptive, `NewVirtualSigstoreWithSigningAlg` or something?